app/code/core/Mage/SalesRule/Model/Resource/Rule/Collection.php

```diff
@@ -91,7 +91,7 @@ public function setValidationFilter($websiteId, $customerGroupId, $couponCode =
                     $connection->quoteInto(
                         'main_table.rule_id = rule_coupons.rule_id AND main_table.coupon_type != ?',
                         Mage_SalesRule_Model_Rule::COUPON_TYPE_NO_COUPON
-                    ),
+                    ) . $connection->quoteInto(' AND rule_coupons.code = ?', $couponCode), // Fixing Performance Issue for Coupon Checkout!
```

Comment:
What happen if `$couponCode = ''`? Has this been tested?

Reply:
the cart page works fine, before and after adding the coupon

Reply:
I just [checked](https://github.com/OpenMage/magento-lts/search?q=setValidationFilter), the method only reference once. So may be we can do a BC change 

from
```php
public function setValidationFilter($websiteId, $customerGroupId, $couponCode = '', $now = null)
```

to
```php
public function setValidationFilter($websiteId, $customerGroupId, $couponCode, $now = null)
```